### ISSUE TITLE
Keep placeholder for empty heading

### ISSUE THREAD

**Issue by Alvol** [quilljs/quill#4122]
After text format changed from `normal` to `heading` placeholder is removed from empty editor
Creating new issue, since #2968 is not fixed in quill 2.0

**Steps for Reproduction**

1. Visit [quill playground](https://quilljs.com/playground/snow)
2. Change format from `Normal` to `Heading 1` or `Heading 2`

**Expected behavior**:
Placeholder should we visible, with heading format applied

**Actual behavior**:
No placeholder

**Platforms**:

Windows 10, Chrome 124, Firefox 126

**Version**:

2.0.0

**Comment by luin** [comment]
Before Quill implements this, you can achieve a similar behavior via pure CSS ([Example](https://quilljs.com/playground/snow#codeN4Ig9AlgdgJgpgDwHQAsAuBbANiAXCAHhggDcACCGAXgB0Q5i0B7AJzoD4CxiT2ap+BAM4BjFhAAOaMkJYja4aPGQArIRy6jxUvlGFoAnlji6yZJAEcsAWgYRmLXCgCGQgBTsyAKlxMoWA2sRFAgsGABKJ1cPbzJPACMWSNx4uAAzVjgyYH4zMxE-NDgoNFwyOgAFAFcWLIBhAGUGsgksZxE4FCYwuBYkfroAbly8gqxWMoBiAE5Z4ag8lqYhewg-Mud4oW6qovmzAF9BMCFDY3YQABpFWEQkNTwQAqhTsgsq0KwyKjIoOAB3MgARQ+WCwbgA5JM7A4IZdsiMMEwYFVjEIyjkFnlmN14s5HGQANojRaE4BkFBwZzwAmEgCM8IATPC0s4sEI4ABdMgHTmXEl5QkQ+LdGBwsgQ+xsiAicUQqq3FhYaBwCF8gVmIUQDDOADmqvhEIK8Gs8XGIgA1mr+VizOqsQcbWYwGAWm0Ol0egSIXUmBgJMsss4FnAJDL+kg4SM0JSMHAyhChFAmP9xS6yKwJfEqvEzar+AdwoMQAcgA)). You have to tweak the style though e.g. increase the placeholder font size if the current format is heading.

**Comment by Alvol** [comment]
@luin yes, thanks for you suggestion. I'm just not sure it's good idea for me to write custom definition of empty text, base on quill inner structure. But, of course, as a workaround it should do it.
